crates/miden-objects/src/testing/storage.rs

```diff
@@ -22,50 +21,55 @@ use crate::utils::sync::LazyLock;
 
 #[derive(Clone, Debug, Default)]
 pub struct AccountStorageDeltaBuilder {
-    values: BTreeMap<StorageSlotName, Word>,
-    maps: BTreeMap<StorageSlotName, StorageMapDelta>,
+    deltas: BTreeMap<StorageSlotName, StorageSlotDelta>,
```

Comment:
nit: `AccountStorageDeltaBuilder` and `AccountStorageDelta` are basically the same structs.
Would it be cleaner to instead add:
```rust
#[cfg(any(feature = "testing", test))]
impl AccountStorageDelta {
   pub fn add_cleared_items(...); 
}
```

etc.? (especially as `AccountStorageDelta` already has a testing impl).

This is just an idea that a) doesn't save that much code duplication; b) might not be the best pattern - but I wanted to flag this, since I just noticed the similarity (and so perhaps others have missed it before too).